target/linux/ramips/dts/R6220.dts

```diff
@@ -0,0 +1,152 @@
+/dts-v1/;
+
+#include "mt7621.dtsi"
+
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/gpio/gpio.h>
+
+/ {
+	model = "Netgear R6220";
+
+	memory@0 {
+		device_type = "memory";
+		reg = <0x0 0x8000000>;
+	};
+
+	chosen {
+		bootargs = "console=ttyS0,115200 maxcpus=2";
+	};
+
+	gpio-leds {
+		compatible = "gpio-leds";
+
+		power {
+			label = "r6220:green:power";
+			gpios = <&gpio0 18 GPIO_ACTIVE_LOW>;
+		};
+		
+		usb {
+			label = "r6220:green:usb";
+			gpios = <&gpio0 15 GPIO_ACTIVE_LOW>;
+		};
+
+		internet {
+			label = "r6220:green:wan";
+			gpios = <&gpio0 13 GPIO_ACTIVE_LOW>;
+		};
+
+		wifi {
+			label = "r6220:green:wifi";
+			gpios = <&gpio0 16 GPIO_ACTIVE_LOW>;
+		};
+
+		wps {
+			label = "r6220:green:wps";
+			gpios = <&gpio0 12 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	gpio-keys-polled {
+		compatible = "gpio-keys-polled";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		poll-interval = <20>;
+
+		wps {
+			label = "wps";
+			gpios = <&gpio0 7 GPIO_ACTIVE_HIGH>;
+			linux,code = <KEY_WPS_BUTTON>;
+		};
+
+		wifi {
+			label = "wifi";
+			gpios = <&gpio0 8 GPIO_ACTIVE_HIGH>;
+			linux,code = <KEY_RFKILL>;
+		};
+
+		reset {
+			label = "reset";
+			gpios = <&gpio0 14 GPIO_ACTIVE_HIGH>;
+			linux,code = <KEY_RESTART>;
+		};
+	};
+
+	gpio_export {
+		compatible = "gpio-export";
+		#size-cells = <0>;
+
+		usbpower {
+			gpio-export,name = "usbpower";
+			gpio-export,output = <1>;
+			gpios = <&gpio0 10 GPIO_ACTIVE_LOW>;
+		};
+	};
+};
+
+&nand {
+	status = "okay";
+
+	partition@0 {
+		label = "u-boot";
+		reg = <0x0 0x100000>;
+		read-only;
+	};
+
+	partition@100000 {
+		label = "SC PID";
+		reg = <0x100000 0x100000>;
+		read-only;
+	};
+
+	partition@200000 {
+		label = "firmware";
```

Comment:
You partitioning is wrong. Boards using NAND (rootfs in ubi) require an partition named ubi.

Depending on whether your bootloader is able to boot a kernel directly from ubi or not, it might be required to add an extra kernel partition as well.

According to your existing image build code, the bootloader doesn't have ubi support and isn't able to boot an kernel from ubi. According to your KERNEL_SIZE variable you expect to have an kernel partition of 4194304 byte.